pkg/rosa/client.go

```diff
@@ -0,0 +1,59 @@
+package rosa
+
+import (
+	"fmt"
+	"os"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+)
+
+type rosaClient struct {
+	ocm *sdk.Connection
+}
+
+// NewRosaClientWithConnection creates a client with a preexisting connection for testing purpose
+func NewRosaClientWithConnection(connection *sdk.Connection) *rosaClient {
+	return &rosaClient{
+		ocm: connection,
+	}
+}
+
+func NewRosaClient(token string) (*rosaClient, error) {
+	ocmAPIUrl := os.Getenv("OCM_API_URL")
+	if ocmAPIUrl == "" {
+		ocmAPIUrl = "https://api.openshift.com"
+	}
```

Comment:
Given that this is a new implementation in tree, can we have the RosaClient here directly read the secret, and eventually fallback to these environment variables?

Reply:
sure